package com.hcl.exception;
/**
 * program for exception.
 * @author Coalesce from vinod.
 */

@SuppressWarnings("serial")
class NumberZeroException extends Exception { 
  /**
  * exception block.
  * @param error for run time arguments.
  */

  public NumberZeroException(String error) { 
    super(error);
  }
}  

class NegativeException extends Exception { 
  public NegativeException(String error) { 
    super(error);
  }
}
public class Custom {
	public void sum(int a,int b) throws NegativeException, NumberZeroException{
		int c;
		boolean flag=true;
		if(a<0||b<0){
			flag=false;
			throw new NegativeException("nagitive not allowed");
		}
		if(a == 0|| b== 0){ 
		flag=false;
		
		throw new NumberZeroException("zero not allowed");
	}
	if(flag==true){
		c=a+b;
		System.out.println(c);
	}
	}
	
public static void main(String[] args) {
	
	int a=5;
	int b=0;
	try {
		new Custom().sum(a, b);
	} catch (NegativeException | NumberZeroException e) {
		// TODO Auto-generated catch block
		e.printStackTrace();
  } 
 } 
}
